libraries/ESP8266WiFiMesh/src/TcpIpMeshBackend.cpp

```diff
@@ -0,0 +1,560 @@
+/*
+  TcpIpMeshBackend
+ 
+  Copyright (c) 2015 Julian Fell and 2019 Anders Löfgren. All rights reserved.
+ 
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <WiFiClient.h> 
+#include <WiFiServer.h>
+#include <assert.h>
+#include <Schedule.h>
+
+#include "TcpIpMeshBackend.h"
+#include "TypeConversionFunctions.h"
+#include "MutexTracker.h"
+
+#define SERVER_IP_ADDR      "192.168.4.1"
+
+const IPAddress TcpIpMeshBackend::emptyIP = IPAddress();
```

Comment:
Why use copy constructor? This should only be used in cases where you know what you're doing, i.e.: custom copy constructors. Prefer default direct constructor instead:
```cpp
const IPAddress TcpIpMeshBackend::emptyIP;
//or
const IPAddress TcpIpMeshBackend::emptyIP();
```

Reply:
Also, consider constexpr instead of const